src/main/java/org/dasein/cloud/digitalocean/compute/DOImage.java

```diff
@@ -201,7 +202,12 @@ protected MachineImage capture(@Nonnull ImageCreateOptions options, @Nullable As
             if( !options.getWithAllRegions() ) {
                 cacheName = regionId;
             }
-            Cache<MachineImage> cache = Cache.getInstance(provider, "images" + ( publicImagesOnly ? "pub" : "prv" ) + "-" + cacheName, MachineImage.class, CacheLevel.REGION_ACCOUNT, new TimePeriod<Minute>(5, TimePeriod.MINUTE));
+            Cache<MachineImage> cache = null;
+            if (publicImagesOnly) {     // PUB
+                cache = Cache.getInstance(provider, "images-pub-" + cacheName, MachineImage.class, CacheLevel.REGION_ACCOUNT, new TimePeriod<Hour>(2, TimePeriod.HOUR) );
+            } else {                    // PRIV
+                cache = Cache.getInstance(provider, "images-prv-" + cacheName, MachineImage.class, CacheLevel.REGION_ACCOUNT, new TimePeriod<Minute>(5, TimePeriod.MINUTE));
+            }
             Collection<MachineImage> cachedImages = ( Collection<MachineImage> ) cache.get(getContext());
             if( cachedImages != null ) {
                 return cachedImages;
```

Comment:
Not your fault, but this code is actually incorrect, because filtering options are not applied here - the whole cached list is returned instead.